Validation/Performance/bin/array.c

```diff
@@ -28,7 +28,7 @@ double **new_Array2D_double(int M, int N) {
 
   if (failed) {
     i--;
-    for (; i <= 0; i--)
+    for (; i >= 0; i--)
```

Comment:
kind of infinite loop for `i=-1` e.g.  when https://github.com/cms-sw/cmssw/blob/master/Validation/Performance/bin/array.c#L17-L23 failed for first item `i=0`.

Reply:
and memory leaks for `i>0`